### PR TITLE
fix(selection): grid row selection and slight refactor 

### PR DESCRIPTION
**Pull request by alebinson**

### CODE REVIEW

**Comment by alebinson** [comment]
Controlling footprint outlineColor/outlineWidth(to avoid filling color) works **ONLY** for **polyline** footprint geojson presentation.

So, multypolygon should be converted to polyline in more accurate way. 

Current situation:
Footprints presented "**as is**" with random fill color

Proposition(by @asafMasa & @shaziri ):
Footprints with no fill but with random border colors and controlled width

Proposed action(by @alebinson ):
1. Approve PR (with review comments fixes)
2. Open an **new** issue to implement requested "transparent footprints" change

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/discrete-layer/components/layers-results/layers-results.tsx [hunk 1]
```diff
@@ -13,24 +16,59 @@ interface LayersResultsComponentProps {
 
 const pagination = true;
 const pageSize = 10;
+const UPDATE_TIMEOUT = 0;
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/layers-results/layers-results.tsx:19]: Done

File: src/discrete-layer/components/layers-results/layers-results.tsx [hunk 2]
```diff
@@ -13,24 +16,59 @@ interface LayersResultsComponentProps {
...
+    } else {
+      cacheRef.current = layersImages;
+      return cacheRef.current;
+    }
+  }
   
   const colDef = [
     {
-      checkboxSelection: true,
+      // checkboxSelection: true,
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/layers-results/layers-results.tsx:63]: Done

File: src/discrete-layer/components/map-container/map-container.tsx [hunk 3]
```diff
@@ -83,6 +83,10 @@ export const MapContainer: React.FC<MapContainerProps> = (props) => {
             type: drawing.type,
           },
         ]);
+
+        if(!resultsOpen){
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/map-container/map-container.tsx:87]: Think it's minor, no need for my opinion

File: src/discrete-layer/components/map-container/selected-layers-container.tsx [hunk 4]
```diff
@@ -1,30 +1,27 @@
...
   const [layersImages, setlayersImages] = useState<ILayerImage[]>([]);
-  const prevLayersImages = usePrevious(layersImages);
+  const prevLayersImages = usePrevious<ILayerImage[]>(layersImages);
   const cacheRef = useRef({} as CacheMap);
   
   useEffect(() => {
     if (discreteLayersStore.layersImages) {
       setlayersImages(discreteLayersStore.layersImages);
+      if(isEmpty(discreteLayersStore.layersImages)) {
+        cacheRef.current = {};
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/map-container/selected-layers-container.tsx:23]: Open an issue https://github.com/MapColonies/discrete-layer-client/issues/50

**Review by asafMasa — APPROVED** [review]

**Pull request merged**